bmt/toolkit.py

```diff
@@ -343,34 +331,63 @@ def get_associations(
             A list of elements
 
         """
-        elements = self.get_descendants("association")
+        association_elements = self.get_descendants("association")
         filtered_elements: List[str] = list()
+        inverse_predicates: Optional[List[str]] = None
+        if predicates:
+            inverse_predicates = list()
+            for pred_curie in predicates:
+                predicate = self.get_element(pred_curie)
+                if predicate:
+                    inverse_p = self.get_inverse(predicate.name)
+                    if inverse_p:
+                        inverse_predicates.append(inverse_p)
+                inverse_predicates = self._format_all_elements(elements=inverse_predicates, formatted=True)
+
+        def match_association(
+                assoc: Element,
+                subj_cats: List[str],
+                preds: List[str],
+                obj_cats: List[str],
+
+        ) -> bool:
+            if subj_cats and not self.match_slot_usage(assoc, "subject", subj_cats):
+                return False
+            if preds and not self.match_slot_usage(assoc, "predicate", preds):
+                return False
+            if obj_cats and not self.match_slot_usage(assoc, "object", obj_cats):
+                return False
+            return True
+
         if subject_categories or predicates or object_categories:
             # This feels like a bit of a brute force approach as an implementation,
             # but we just use the list of all association names to retrieve each
             # association record for filtering against the constraints?
-            for name in elements:
+            for name in association_elements:
+
                 association: Optional[Element] = self.get_element(name)
+                if not association:
+                    continue
```

Comment:
above, you look for descendants of "association", I like the double check here that you can make the retrieved name into an element, can it be combined with the call above and remove the "if not association continue"  line here (since it will be an association)?  or did you find places in the model where an association descends from a mixin?  (those we should fix)

Reply:
Hmm...
```
association_elements = self.get_descendants("association")
```
probably returns valid Elements for every entry, so the extra test here is probably unnecessary (a NOP). I guess I was mindlessly lured into doing the test against the result since self.get_element(name) returns Optional[Element] LOL